### ISSUE TITLE
Running a worker script through a worker and hangs if there is a Deno.mkdir() in the worker script

### ISSUE THREAD

**Issue by skanehira** [denoland/deno#19455]
# Summary
Running a worker script through a worker and hangs if there is a Deno.mkdir() in the worker script.
You can reproduce this using https://github.com/lambdalisue/deno-hung-test

## Details
The process flow is as follows

1. `test.ts` spawn Worker with `worker.ts`
2. `worker.ts` dynamically import `plugin.ts` and invoke main function of it
4. `plugin.ts` invoke await `Deno.mkdir(...)` when loaded

When `deno run -A test.ts`, `plugin.ts`'s loading would not finised.

```sh
$ deno run -A ./test.ts
[plugin] before mkdir
[plugin] after mkdir
[plugin] loaded
```

But when delete `await Deno.mkdir()` loading would be success.

```sh
$ deno run -A ./test.ts
[plugin] before mkdir
[plugin] after mkdir
[plugin] loaded
[worker] mod loaded
[worker] before main
[plugin] success
[worker] after main
```

## Environments
```sh
$ deno --version
deno 1.34.2 (release, aarch64-apple-darwin)
v8 11.5.150.2
typescript 5.0.4
```

**Comment by hashue** [comment]
This issue happend `v1.32.4` and over
```
$ deno -V
deno 1.32.3
$ deno run -A test.ts
[plugin] before mkdir
[plugin] after mkdir
[plugin] loaded
[worker] mod loaded
[worker] before main
[plugin] success
[worker] after main

$ deno -V
deno 1.32.4
 $ deno run -A test.ts
[plugin] before mkdir
[plugin] after mkdir
[plugin] loaded
$
```

**Comment by lambdalisue** [comment]
It's a bit difficult but if we move `await Deno.mkdir(...)` into `main()` function, it works. This issue happens only when `await Deno.mkdir(...)` is called on script root.

**Comment by lambdalisue** [comment]
Found it. https://github.com/denoland/deno/commit/e222c368c0d26f4af240b316df3bde26c9f054d1 is the reason. I revert that commit on deno 1.32.4 and the issue is resolved.

@magurotuna I'm not sure if this can be just reverted so what should we do next?

**Comment by magurotuna** [comment]
I don't 100% know the context of the original PR either, but generally it would be good to revert with regression test cases added in my opinion.

**Comment by bartlomieju** [comment]
@lambdalisue I'm not 100% sure if we can revert the PR. I'll try and let you know.

**Comment by mmastrac** [comment]
This might be the same as [19823](https://github.com/denoland/deno/issues/19823).

Note that it's failing in a different way now:

```
[worker] load plugin
[plugin] before mkdir
[plugin] after mkdir
[plugin] loaded
(exits)
```

**Comment by bartlomieju** [comment]
Thinking about it more - my hunch is that cross-thread wake ups are not working correctly anymore in this situation. We're trying to come up with a reproducible test case in `deno_core`.

**Comment by mmastrac** [comment]
Based on the info from this and #19903, I managed to track it down to pending ops blocking top-level awaits from resolving.

```ts
// The existence of a pending op should not prevent top-level module awaits from resolving

// https://github.com/denoland/deno/issues/19903
// https://github.com/denoland/deno/issues/19455

// If this timer triggers, we locked up
const timeout = setTimeout(() => {
  Deno.exit(1);
}, 10000);

// This will not resolve in time, but it shouldn't block top-level awaits
(async () => {
  const timeout = setTimeout(() => null, 100000);
  Deno.unrefTimer(timeout);
  await new Promise(_ => null);
})();

const moduleCode = `
console.log('module start');
const hash = await crypto.subtle.digest('SHA-1', new TextEncoder().encode('data'));
console.log('module finish');
`;
console.log('before import');
await import(URL.createObjectURL(new Blob([ moduleCode ])));
console.log('after import');

clearTimeout(timeout);
```

**Comment by mmastrac** [comment]
Fixed in 1.35.3
